arviz/tests/base_tests/test_data.py

```diff
@@ -323,6 +324,18 @@ def test_addition(self):
         fails = check_multiple_attrs(test_dict, new_idata)
         assert not fails
 
+    def test_extend_xr_method(self):
+        arr = xr.DataArray(
+            data=np.ones((2, 3)),
+            dims=["x", "y"],
+            coords={"x": range(2), "y": range(3), "a": ("x", [3, 4])},
+        )
+        ds = xr.Dataset({"v": arr})
+        dataset = convert_to_inference_data(ds)
+        assert_identical(dataset.set_index(x="a").posterior, ds.set_index(x="a"))
+        dataset.set_index(x="a", inplace=True)
+        assert_identical(dataset.posterior, ds.set_index(x="a"))
```

Comment:
kudos on testing both inplace values, however, I think it would make the test easier to follow if a fixture was used for that. `test_sel` can be a template on `inplace`, not so much on the helper I was talking about above.